integrations/gen_docs_integrations.py

```diff
@@ -1,10 +1,32 @@
 import json
 import os
+import shutil
 
 # Dictionary responsible for making the symbolic links at the end of the script's run.
 symlink_dict = {}
+am_i_inside_go = "go.d.plugin" in os.getcwd()
 
 
+def cleanup():
+    """
+    clean directories that are either data collection or exporting integrations
+    """
+    if am_i_inside_go:
+        for subdir, dirs, files in os.walk("modules"):
+            for directory in dirs:
+                if directory == "integrations":
+                    shutil.rmtree(os.path.join(subdir, directory))
```

Comment:
```suggestion
        for subdir, dirs, files in os.walk("modules"):
            for directory in dirs:
                if directory == "integrations":
                    shutil.rmtree(os.path.join(subdir, directory))
                    break
```

Because you’re looking for an exact match here and directory entries must be unique within a given directory, you can break out of the innermost loop as soon as you find that match to save a bit of time.